### PR TITLE
config file updated.

### PR DESCRIPTION
**Pull request by susmita-debnath**

### CODE REVIEW

File: app/controllers/AjaxController.php [hunk 1]
```diff
@@ -19,11 +19,17 @@ public function get_data() {
...
+				$this->mobile = $_POST['mobile'];
 				$this->gender = $_POST['gender'];
 				$this->address = $_POST['address'];
 
 				print_r($this);
 			}
+
+			public function register() {
+				$user = new User();
+				$user->name =
```
Comment (bappi-d-great) [inline app/controllers/AjaxController.php]: This should show an error: `$user->name = ` Why does no value assigned to it?

File: app/controllers/User.php [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 		class User {
 			
+			
+			
 			function __construct() {
 
 			}
```
Comment (bappi-d-great) [inline app/controllers/User.php]: What is this file for?

File: app/lib/functions.php [hunk 3]
```diff
@@ -5,4 +5,16 @@
...
+
+	if ( ! function_exists( 'base_url' ) ) {
+		function base_url() {
+			return BASE_URL;
+		}
+	}
+
+	if ( ! function_exists( 'get_table' ) ) {
+		function get_table() {
+			return TABLE_PREFIX;
```
Comment (bappi-d-great) [inline app/lib/functions.php]: `get_table()` should return the table name, not the prefix. This function should have a param, then add the prefix with the param and return as the table name.

File: config.php [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+		if ( ! defined( 'BASE_URL' ) ) define('BASE_URL', 'bappi-d-great.com/hospital');
+	    if ( ! defined( 'TABLE_PREFIX' ) ) define('TABLE_PREFIX', 'hp_');
+	}
+	else {
+		if ( ! defined( 'DB_HOST' ) ) define( 'DB_HOST', 'localhost' );
+		if ( ! defined( 'DB_USER' ) ) define( 'DB_USER', 'root' );
+		if ( ! defined( 'DB_PASSWORD' ) ) define( 'DB_PASSWORD', '' );
+		if ( ! defined( 'DB_NAME' ) ) define( 'DB_NAME', 'hospital' );
+		if ( ! defined( 'BASE_URL' ) ) define('BASE_URL', 'hospital.test');
+		if ( ! defined( 'TABLE_PREFIX' ) ) define('TABLE_PREFIX', 'hp_');
```
Comment (bappi-d-great) [inline config.php]: `if ( ! defined( 'TABLE_PREFIX' ) ) define('TABLE_PREFIX', 'hp_');`

This line is same on both conditions, don't need to write twice.

**Review by bappi-d-great — CHANGES_REQUESTED** [review]
Need some fix.

File: app/controllers/AjaxController.php (hunk 1 above)
Comment (susmita-debnath) [reply app/controllers/AjaxController.php]: Because I did that much for yesterday :(

File: app/controllers/AjaxController.php [hunk 5]
```diff
@@ -19,10 +19,20 @@ public function get_data() {
...
+				//print_r($this);
+			}
+
+			public function register() {
+				$user = new User();
+				if ( true === $user->uniquenessChecking($_POST['email']) ) {
+					$user->set_user_data( $_POST );
+					$user->save();
+				}
+				else echo "The email has already been taken.";
```
Comment (bappi-d-great) [inline app/controllers/AjaxController.php]: There should be an `exit()` function after echo to terminate the process after ajax response.

File: app/lib/functions.php [hunk 6]
```diff
@@ -5,4 +5,28 @@
...
+		function get_table( $name ) {
+			return TABLE_PREFIX . $name; 
+		}
+	}
+
+	if ( ! function_exists( 'get_activation_key' ) ) {
+		function get_activation_key( $email, $name ) {
+			$salt1 = 'sdjflsaf';
+			$salt2 = 'askfjsfa';
+			$salt3 = 'skjfksjdfajkjd';
```
Comment (bappi-d-great) [inline app/lib/functions.php]: Use number and special characters too as salt keys.

File: assets/js/src/register.js [hunk 7]
```diff
@@ -49,7 +49,9 @@ jQuery( document ).ready( function($){
 
 	$('#datepicker').datepicker({
         changeYear: true,
-        yearRange: "1900:2020"
+        yearRange: "1900:2020",
```
Comment (bappi-d-great) [inline assets/js/src/register.js]: If year range is up to 2020? What will be in 2021?

File: assets/js/src/register.js [hunk 8]
```diff
@@ -59,7 +61,7 @@ jQuery( document ).ready( function($){
 			pwerr += '<li>Minimum 1 uppercase letter & 1 lowercase letter </li>';
 			pwerr += '<li>Minimum 1 special character</li>';
 		pwerr += '</ul>';
-	var errors = ['<p>All fields are required.</p>', '<p>Email is invalid.</p>', pwerr, '<p>Password is not same</p>'];
+	var errors = ['<p>All fields are required.</p>', '<p>Email is invalid.</p>', pwerr, '<p>Password is not same</p>', '<p>The email has already been taken.</p>'];
```
Comment (bappi-d-great) [inline assets/js/src/register.js]: The error message for email exist shouldn't be here.

File: assets/js/src/register.js [hunk 9]
```diff
@@ -103,18 +110,25 @@ jQuery( document ).ready( function($){
...
 				address: address
 			},
 			success: function( response ) {
 				console.log( response );
+				$('#error').html(response);
+				$('#error').addClass('alert alert-danger');
+
+				$('html, body').animate({
+					scrollTop: $('#error').offset().top
+				});
```
Comment (bappi-d-great) [inline assets/js/src/register.js]: - Why is there `console.log`?
- Why it's always considering as error? What if the user is successfully registered?

**Review by bappi-d-great — CHANGES_REQUESTED** [review]

**Pull request closed without merging**